c++/src/ColumnReader.cc

```diff
@@ -327,7 +329,9 @@ namespace orc {
             nanoBuffer[i] *= 10;
           }
         }
-        secsBuffer[i] += epochOffset;
+        int64_t writerTime = secsBuffer[i] + epochOffset;
+        secsBuffer[i] = writerTime +
```

Comment:
@omalley I think here introduces an inconsistency between reader and writer. The secsBuffer[i] in the TimestampVectorBatch of reader side will not get same value on the java writer/reader side.